sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -486,18 +486,8 @@ case class AlterTableAddPartitionCommand(
     }
 
     sparkSession.catalog.refreshTable(table.identifier.quotedString)
-    if (table.stats.nonEmpty) {
-      if (sparkSession.sessionState.conf.autoSizeUpdateEnabled) {
-        val addedSize = CommandUtils.calculateMultipleLocationSizes(sparkSession, table.identifier,
```

Comment:
I think this avoids the external catalog call if `addedSize` is 0, which `updateTableStats` doesn't do, but I guess in most cases `addedSize` should be > 0.

Reply:
You are right. I think we could do similar optimization in `updateTableStats` by comparing new size with existing one in `CatalogTable`. So, the optimization can be picked up by other commands.

Reply:
can we do that first? otherwise this PR may introduce perf regression.

Reply:
Here is the PR https://github.com/apache/spark/pull/31135